os/kernel/pthread/pthread_tryjoin_np.c

```diff
@@ -0,0 +1,239 @@
+/****************************************************************************
+ *
+ * Copyright 2018 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ *
+ *   Copyright (C) 2007, 2008, 2011, 2013 Gregory Nutt. All rights reserved.
+ *   Author: Gregory Nutt <gnutt@nuttx.org>
```

Comment:
Is it back ported from NuttX?, if not shall we remove this?

Reply:
Actually no. But it is almost same as phtread_join. The difference between pthread_join and pthread_tryjoin_np is blockable. When thread is not terminated, pthread_join blocks this context. But pthread_tryjoin_np just returns with error. So, I just remove blocking logic from pthread_join. Except removing, all are same as pthread_join. That's why I left it.